tests/countries/test_sierra_leone.py

```diff
@@ -0,0 +1,180 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries import SierraLeone, SL, SLE
+from tests.common import CommonCountryTests
+
+
+class TestSierraLeone(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1962, 2050)
+        super().setUpClass(SierraLeone, years=years, years_non_observed=years)
+        cls.no_estimated_holidays = SierraLeone(years=years, islamic_show_estimated=False)
+
+    def test_country_aliases(self):
+        self.assertAliases(SierraLeone, SL, SLE)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(SierraLeone(years=1961))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1962, 2050)))
+
+    def test_armed_forces_day(self):
+        name = "Armed Forces Day"
+        self.assertHolidayName(name, (f"{year}-02-18" for year in range(2002, 2050)))
+        self.assertNoHolidayName(name, range(1962, 2002))
+
+    def test_international_womens_day(self):
+        name = "International Women's Day"
+        self.assertHolidayName(name, (f"{year}-03-08" for year in range(2018, 2050)))
+        self.assertNoHolidayName(name, range(1962, 2018))
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(name, "2022-04-15", "2023-04-07", "2024-03-29")
+        self.assertHolidayName(name, range(1962, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(name, "2022-04-18", "2023-04-10", "2024-04-01")
+        self.assertHolidayName(name, range(1962, 2050))
```

Comment:
Initially, there were good tests here, in which only future dates were redundant.